modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -134,8 +147,72 @@ function AddFDSFilterModalContent({
 				);
 			}
 		});
+
+		if (filter?.filterType === EFilterType.SELECTION) {
+			const selectionFilter = filter as ISelectionFilter;
+			setSourceType(selectionFilter.sourceType);
+		}
 	}, [filter]);
 
+	const isInvalidForm = ({
```

Comment:
[non-blocking nit] Naming here is off. 
1. The most significant word should be on the right. Then we add descriptors on the left, that reduce scope, on the left.
2. A boolean or a function that returns a boolean should have an adjective as the most significant word.

With that, we go from `isValid()` to `isFormValid()`, or if more specificity is needed something like `isNewDateRangeFilterFormValid()`. `Invalid` is fine, although a tad unusual. So, in this case, `isFormInvalid` would be ok.